tests/foreman/ui_airgun/test_discoveredhost.py

```diff
@@ -0,0 +1,496 @@
+@tier3
+@upgrade
+def test_positive_pxe_less_with_dhcp_unattended(
+        session, module_org, module_loc, provisioning_env):
+    """Discover a host with dhcp via bootable discovery ISO by setting
+    "proxy.type=proxy" in PXE default in unattended mode.
+
+    :id: fc13167f-6fa0-4fe5-8584-7716292866ce
+
+    :Setup: Provisioning should be configured
+
+    :Steps: Boot a host/VM using modified discovery ISO.
+
+    :expectedresults: Host should be successfully discovered
+
+    :CaseLevel: System
+    """
+    with LibvirtGuest(boot_iso=True) as pxe_less_host:
+        host_name = pxe_less_host.guest_name
+        with session:
+            session.organization.select(org_name=module_org.name)
+            session.location.select(loc_name=module_loc.name)
+            discovered_host_values = session.discoveredhosts.wait_for_entity(
+                host_name)
+            assert discovered_host_values['Name'] == host_name
+
+
+@run_in_one_thread
+@run_only_on('sat')
+@skip_if_not_set('compute_resources', 'vlan_networking')
+@tier3
+@upgrade
+def test_positive_provision_using_quick_host_button(
+        session, module_org, module_loc, provisioning_env):
+    """Associate hostgroup while provisioning a discovered host from
+    host properties model window and select quick host.
+
+    :id: 34c1e9ea-f210-4a1e-aead-421eb962643b
+
+    :Setup:
+
+        1. Host should already be discovered
+        2. Hostgroup should already be created with all required entities.
+
+    :expectedresults: Host should be quickly provisioned and entry from
+        discovered host should be auto removed.
+
+    :CaseLevel: System
+    """
+    with LibvirtGuest() as pxe_host:
+        host_name = pxe_host.guest_name
+        with session:
+            session.organization.select(org_name=module_org.name)
+            session.location.select(loc_name=module_loc.name)
+            discovered_host_values = session.discoveredhosts.wait_for_entity(
+                host_name)
+            assert discovered_host_values['Name'] == host_name
+            session.discoveredhosts.provision(
+                host_name,
+                provisioning_env['host_group'],
+                module_org.name,
+                module_loc.name,
+            )
+            pxe_host_name = '{0}.{1}'.format(
+                host_name, provisioning_env['domain'])
+            # After this step we will be redirected to the created host details
+            # view.
+            values = session.host.get_details(pxe_host_name)
+            assert values['properties']['properties_table']['Status'] == 'OK'
+            # Check that provisioned host is not in the list of discovered
+            # hosts anymore
+            assert not session.discoveredhosts.search(
+                'name = {0}'.format(host_name))
+
+
+@run_in_one_thread
+@run_only_on('sat')
+@skip_if_not_set('compute_resources', 'vlan_networking')
+@tier3
+def test_positive_update_name(
```

Comment:
this is what my original comment was about - let;s simply create a `make_fake_discovery_host` factory method and use it here (we use similar `make_fake_host` already in other host-related tests).
This will instantly make the test a T2 tests which will be much faster and will test the same thing - note that we are not interacting with the real host here in any way. We simply create the host jsut so it shows up in the discovered hosts - but we can mock this easily.
- if you want, i can write the factory for you.

Reply:
@rplevka How we can provision a fake host?
Note: we are provisioning by updating the discovered host, provisioning data by changing only the name.

Please clarify: """we use similar make_fake_host already in other host-related tests""", in which test we are using fake host for provisioning.

Reply:
@rplevka I understand that we can create a test discovered host but we are in tier3 and we need a real user scenario. Do not see why we have to cut some tests as soon as we have already all the setups already done in this module, the run time is adequate here, but the value is very high.

Reply:
@ldjebran because we are not interested in HOW the discovered host got into satellite here. the test says that the host should be already discovered and it only tries the renaming part.
this should not be tier3 at all (see my first comment on this, which suggests to make this tier2).

- we have the proper discovery covered in other tests.
- making the full-blown discovery just to test we can do some CRUD stuff on its result is plain stupid

Reply:
@rplevka this definitely not working with test discovered host as when provisioning the system want to reboot that host.

Reply:
@ldjebran we don't do that in this test. Anyway, even if we did, UI handles this just fine and creates new host record from a fake discovered host just fine.

Reply:
suppose we have created a fake discovered host and we have this test code
```python
def test_positive_update_name(
        session, module_org, module_loc, module_host_group, discovered_host):
    """Update the discovered host name and provision it

    :id: 3770b007-5006-4815-ae03-fbd330aad304

    :Setup: Host should already be discovered

    :expectedresults: The hostname should be updated and host should be
        provisioned

    :CaseLevel: System
    """
    discovered_host_name = discovered_host['name']
    domain_name = module_host_group.domain.read().name
    new_name = gen_string('alpha').lower()
    new_host_name = '{0}.{1}'.format(new_name, domain_name)
    with session:
        session.organization.select(org_name=module_org.name)
        session.location.select(loc_name=module_loc.name)
        discovered_host_values = session.discoveredhosts.wait_for_entity(
            discovered_host_name)
        assert discovered_host_values['Name'] == discovered_host_name
        session.discoveredhosts.provision(
            discovered_host_name,
            module_host_group.name,
            module_org.name,
            module_loc.name,
            quick=False,
            host_values={'host.name': new_name}
        )
        assert (session.host.search(new_host_name)[0]['Name']
                == new_host_name)
        values = session.host.get_details(new_host_name)
        assert values['properties']['properties_table']['Status'] == 'OK'
        assert not session.discoveredhosts.search(
            'name = {0}'.format(discovered_host_name))
```
The test will fail in provisioning when trying to reboot the host here is a screen shot
![screenshot from 2018-10-23 13-30-22](https://user-images.githubusercontent.com/131553/47369456-64688f00-d6ec-11e8-8cd9-1579d9089c13.png)